RationaleMCP/0035/SpecificationText.md

```diff
@@ -0,0 +1,84 @@
+# 13.6 Multilingual Descriptions
+
+[*Descriptive texts in a model or library are usually formulated in English. To support users who are not so familiar with this, a Modelica tool can support the translation of these texts into any language. This should be done using external files, so that no modification of the actual Modelica text is required.*]
+
+The files to support translation must be provided by the developer of the library. They must be stored in a subdirectory of the Resources directory of the library with the name `Language`.
+
+Two kind of files have to be provided:
+* a template file `<LibraryName>.pot`. 
+It contains all necessary information to translate all descriptions, but no translations. The pattern `<LibraryName>` denotes the toplevel class name of the library.
+* one file for each supported language with the name `<LibraryName>.<language>.po`. This file is a copy of the associated template file, but extended with the translations in the specified language. The pattern `<language>` stands for the ISO 639-1 language code, e.g. de or en.

```

Comment:
Is it a deliberate decision to not support the binary _.mo_ file format? If so, I think it would be wise to state it clearly.

Reply:
I would say it is a deliberate decision, since we don't want to standardize the implementation (especially not the gettext-specific hashing).

Reply:
Right, let's add a normative statement about that, and see if it causes any reactions under the MCP evaluation.  Opening up for allowing the _.mo_ format later would of course be easy if we'd ever change our mind (for example, it it turns out that all tools are using the LGPL getttext library anyway in the end).

Reply:
By the way, the way I imagine the Modelica library _.po_ files would be used in System Modeler is that the tool has a cache of _.mo_ files that are created when the library is loaded (the gettext _msgfmt_ program (GPL) can do this), so that the translated strings can then be accessed efficiently using the widely available _libintl_ (LGPL) library.  If anyone sees any catch with this approach, please bring it to light.

Reply:
To me creating a binary file is something that tools may optionally add (in that format or some other one), but the specification standardizes on the ".po" file so that all tools can translate the library in the same way.

Reply:
> To me creating a binary file is something that tools may optionally add (in that format or some other one), but the specification standardizes on the ".po" file so that all tools can translate the library in the same way.

I agree that this is the right way to go, which is why I am suggesting to add a non-normative comment to clarify that a library cannot be distributed with _.mo_-files as a substitute for the _.po_-files.

Reply:
I added a generalization of that as a normative text; https://github.com/modelica/ModelicaSpecification/pull/2956#discussion_r983395460_ - resolving this to continue the discussion there.